src/main/java/Duke.java

```diff
@@ -1,10 +1,152 @@
+//package duke;
+import java.io.IOException;
+import java.util.*;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FileNotFoundException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+/**
+ * Duke the best chatbot hehe
+ */
 public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private Parser parser;
+
+    /**
+     * Init Duke
+     */
+    public Duke() {
+        ui = new Ui();
+        storage = new Storage();
+        tasks = new TaskList(storage.readData());
+        parser = new Parser();
+    }
+
+    /**
+     * Main
+     */
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        new Duke().run();
+    }
+
+    /**
+     * Take in command
+     */
+    public void run() {
+        ui.sayHi();
+        Scanner myScanner = new Scanner(System.in);
+        while(true) {
+            String cmd = myScanner.nextLine();
+            if(cmd.equals("bye")) {
+                ui.sayBye();
+                break;
+            }
+            else if(cmd.equals("list")) {
+                System.out.println("Here are the tasks in your list:");
+                for(int i = 1; i <= tasks.getSize(); ++i) {
+                    System.out.println(i + "." + tasks.get(i - 1).getStatus());
+                }
+            }
+            else if(cmd.length() >= 4 && cmd.substring(0, 4).equals("done")) {
```

Comment:
Perhaps you could use comments to further elaborate on your "else if" statements' condition? I feel like it would definitely be clearer and will increase readability for readers!